pkg/virt-controller/watch/vm.go

```diff
@@ -712,8 +711,7 @@ func (c *VMController) VMIAffinityPatch(vm *virtv1.VirtualMachine, vmi *virtv1.V
 	var ops []string
 
 	if vm.Spec.Template.Spec.Affinity != nil {
-		vmAffinity := vm.Spec.Template.Spec.Affinity.DeepCopy()
-		vmAffinityJson, err := json.Marshal(vmAffinity)
+		vmAffinityJson, err := json.Marshal(vm.Spec.Template.Spec)
```

Comment:
You seem to be marshaling the whole `Spec` and not the affinity.

I guess this area is not unit tested.

Reply:
@xpivarc Functional test fails because of this change here.

Reply:
Thanks, I was too eager with the delete button.